env/work/srw_python/srwlib.py

```diff
@@ -4271,6 +4272,19 @@ def srwl_wfr_emit_prop_multi_e(_e_beam, _mag, _mesh, _sr_meth, _sr_rel_prec, _n_
         #iAvgProc += 1 #OC190414 (commented-out)
         #iSave += 1
 
+    #MR01112016: Initialize prarameters for the SRW status files and generate the files:
+    log_dir = os.getcwd() if _file_path is None else os.path.dirname(os.path.abspath(_file_path))
+    timestamp = '{:%Y-%m-%d_%H-%M-%S}'.format(datetime.datetime.now())
+    log_file = 'srw_status_{}'.format(timestamp)
+    log_path = os.path.join(log_dir, log_file)
+    if nProc <= 1:
+        total_num_of_particles = nPartPerProc
+    else:
+        total_num_of_particles = nSentPerProc * (nProc - 1) * _n_part_avg_proc
+    if rank == 0:
+        srwl_uti_save_stat_wfr_emit_prop_multi_e(0, total_num_of_particles, filename=log_path,
+                                                 cores=nProc, particles_per_iteration=_n_part_avg_proc)
+
```

Comment:
These things should be done in a separate function (perhaps in srwl_uti_save_stat_wfr_emit_prop_multi_e, or in another one of this kind, e.g. "srwl_uti_save_stat_wfr_emit_prop_multi_e_init" or something) that should be called from here or maybe from the very beginning of "srwl_wfr_emit_prop_multi_e". One can't mix-up these auxiliary actions with partially-coherent radiation calculation algorithm; "srwl_wfr_emit_prop_multi_e" is already very complicated, even without this...